great_expectations/data_context/data_context.py

```diff
@@ -3348,6 +3348,27 @@ def run_profiler_with_dynamic_arguments(
             include_citation=include_citation,
         )
 
+    @usage_statistics_enabled_method(
+        event_name="data_context.run_profiler_on_data",
+    )
+    def run_profiler_on_data(
+        self,
+        batch_request: Union[dict, BatchRequest, RuntimeBatchRequest],
+        name: Optional[str] = None,
+        ge_cloud_id: Optional[str] = None,
+        expectation_suite_name: Optional[str] = None,
+        include_citation: bool = True,
```

Comment:
Added all these args since they are passed downstream but let me know if this external-facing API should be a bit more limited.

Reply:
I think my only concern here would be `ge_cloud_id`, but then it's already passed into other methods in the `DataContext`-level

Reply:
@cdkini This seems fine -- the key ones for us are in there...  I do wonder if we also need to include the `cloud_id` for the expectation_suite... not sure -- perhaps ask someone on that team.

Reply:
Will leave for now but revisit with Cloud as needed!